### DISCUSSION TITLE
Capturing MAC packets with srsUE over the air in real time

### DISCUSSION THREAD

**Discussion by no-tech-no** [srsran/srsRAN_4G#1329] [Q&A]
Hello,

Our student group is currently utilizing the [BladeRF 2.0 micro xA9](https://www.nuand.com/product/bladerf-xa9/) for our project. We have successfully established our own 5G NSA and 4G network using [Open5Gs](https://open5gs.org/).

[Objective]
Our aim is to actively capture the traffic between two BladeRF devices (UE and GnodeB) using a third BladeRF device possibly running a customized version of srsUE.

[Inquiries]
1) How could analyse the traffic in real time with Wireshark? Is srsUE the appropriate tool for this purpose?
2) What level of complexity is involved in modifying srsUE for this specific use case?

**Comment by ryantheelder** [comment]
Hi there,

Sniffing 5G/4G packets over the air is an active area of research, and I'd recommend looking into a couple of key articles that could provide valuable information and context for your project:

1. [LTESniffer](https://github.com/SysSec-KAIST/LTESniffer) (Thanks to @SysSec-KAIST Team)
2. [5GSniffer](https://github.com/spritelab/5GSniffer) (Thanks to @spritelab Team)
